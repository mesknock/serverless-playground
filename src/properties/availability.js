import { successResponse, runWarm } from '../utils';

const availability = async (event, context) => {
  return successResponse({ foo: 'bar' });
};

// runWarm function handles pings from the scheduler so you don't
// have to put that boilerplate in your function.
export default runWarm(availability);
