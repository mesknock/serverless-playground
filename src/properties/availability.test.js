import availability, { isValid } from './availability';

describe('availability', () => {
  it('executes as expected', async () => {
    expect(
      await availability({ pathParameters: { propertyId: '1234' } }, {})
    ).toMatchSnapshot();
  });

  it('is invalid', async () => {
    expect(
      await availability({ pathParameters: { propertyId: 'aaaa' } }, {})
    ).toMatchSnapshot();
    // TODO: ...
  });
  it('cannot be 123', async () => {
    expect(
      await availability({ pathParameters: { propertyId: '123' } }, {})
    ).toMatchSnapshot();
    // TODO: ...
  });
});
