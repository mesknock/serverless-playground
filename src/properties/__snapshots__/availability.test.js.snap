// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`availability cannot be 123 1`] = `
Object {
  "body": "{\\"errorMessage\\":\\"cannot be 123\\"}",
  "statusCode": 400,
}
`;

exports[`availability executes as expected 1`] = `
Object {
  "body": "{\\"foo\\":\\"bar\\",\\"propertyId\\":\\"1234\\"}",
  "statusCode": 200,
}
`;

exports[`availability is invalid 1`] = `
Object {
  "body": "{\\"errorMessage\\":\\"yup says no\\"}",
  "statusCode": 400,
}
`;
